cmd/proxy/main.go

```diff
@@ -80,6 +81,21 @@ func main() {
 		}
 	}()
 
+	//agent check-in routine
```

Comment:
[Session.CloseChan](https://pkg.go.dev/github.com/hashicorp/yamux?utm_source=godoc#Session.CloseChan) should be used instead of a for loop

Reply:
I don't understand exactly what do U mean about CloseChan and how it can be coded..